placehero/src/avatars.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use xilem::core::one_of::Either;
+use xilem::core::{MessageProxy, NoElement, View};
+use xilem::palette::css;
+use xilem::style::{Gradient, Style};
+use xilem::tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+use xilem::view::{image, sized_box, spinner, worker};
+use xilem::{Blob, Image, ImageFormat, ViewCtx, WidgetView, tokio};
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[derive(Debug)]
+struct AvatarRequest {
+    avatar_url: String,
+}
+
+#[derive(Debug)]
+struct AvatarResponse {
+    url: String,
+    image: Image,
+}
+
+#[derive(Default)]
+pub(crate) struct Avatars {
+    icons: HashMap<String, Option<Image>>,
+    requester: Option<UnboundedSender<AvatarRequest>>,
+    // TODO: lru: ...
+}
+
+impl Avatars {
+    pub(crate) fn worker(&mut self) -> impl View<Self, (), ViewCtx, Element = NoElement> + use<> {
```

Comment:
We probably want an trait "alias" for `NoElement` views to avoid this don't we?

Reply:
Opened #1093